pandas/core/categorical.py

```diff
@@ -313,7 +322,8 @@ def __init__(self, values, categories=None, ordered=False, fastpath=False):
                 raise NotImplementedError("> 1 ndim Categorical are not "
                                           "supported at this time")
 
-            categories = self._validate_categories(categories)
+            if dtype is None or isinstance(dtype, str):
+                dtype = CategoricalDtype(categories, ordered)
```

Comment:
make this ``CategoricaDtype.construct_from_string(dtype, categories=categories, ordered=ordered)`` I think (just for validation in case someone passes

```
Categorical(....., dtype='foo')
```

Reply:
I think that if a user passes `dtype='foo'`, this actually already raises before (as in the beginning of init the .categories and .ordered attributes of the dtype are accessed if it is not None). 
Might be good to actually have a test case for this.

Reply:
did you mean Categorical instead of CategoricalDtype ?

Reply:
I meant exactly what I wrote.

the handling of ``dtype=`` in ``Categorical`` is duplicative and just needs to pass directly to ``CategoricalDtype`` (and that's why we need a constructor ``.from_string``)

Reply:
``CategoricalDtype([1, 2], dtype='categories')`` this should raise

CategoricalDtype([1, 2], dtype='category') might be ok though

Reply:
Whoops, I don't know what I was thinking when I wrote https://github.com/pandas-dev/pandas/pull/16015/files#r139390429

Here's the behavior (only `dtype=CategoricalDtype(...)` or `dtype='category'` works):

```python
In [2]: pd.Categorical([1, 2])
Out[2]:
[1, 2]
Categories (2, int64): [1, 2]

In [3]: pd.Categorical([1, 2], dtype='category')
Out[3]:
[1, 2]
Categories (2, int64): [1, 2]

In [4]: pd.Categorical([1, 2], dtype='categories')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-480c8cb555e4> in <module>()
----> 1 pd.Categorical([1, 2], dtype='categories')

~/Envs/pandas-dev/lib/python3.6/site-packages/pandas/pandas/core/categorical.py in __init__(self, values, categories, ordered, dtype, fastpath)
    240                     dtype = CategoricalDtype(categories, ordered)
    241                 else:
--> 242                     raise ValueError("Unknown `dtype` {}".format(dtype))
    243             elif categories is not None or ordered is not None:
    244                 raise ValueError("Cannot specify both `dtype` and `categories`"

ValueError: Unknown `dtype` categories
``

What's the argument for `CategoricalDtype(..., dtype='category')`? At the moment, I haven't added a dtype to `CategoricalDtype.__init__`. I'm not sure whether or not to do it, and if we do accept `CategoricalDtype` in `CategoricalDtype.__init__` then should it be the first argument `categories`, or a separate keyword? I'm going through the PR again, and I'll note all the places I have to do `isinstance(foo, CategoricalDtype)` before passing it to the constructor.

I'm also not sure changing `CategoricalDtype.construct_from_string` to accept categories (is that what you were suggisting @jreback?). Currently, it's just

```python
In [12]: pd.api.types.CategoricalDtype.construct_from_string('category')
Out[12]: CategoricalDtype(categories=None, ordered=False)
```

Everything else raises.